specification/marketplace/resource-manager/Microsoft.Marketplace/stable/2020-12-01/Marketplace.json

```diff
@@ -0,0 +1,897 @@
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "UpdateOfferNotification": {
+            "$ref": "./examples/UpdateOfferNotification.json"
+          }
+        }
+      }
+    },
+    "/providers/Microsoft.Marketplace/privateStores/{privateStoreId}/offers/{offerId}/requestApprovals": {
+      "parameters": [
+        {
+          "$ref": "#/parameters/PrivateStoreIdParameter"
+        },
+        {
+          "$ref": "#/parameters/OfferIdPathParameter"
+        },
+        {
+          "$ref": "#/parameters/ApiVersionParameter"
+        }
+      ],
+      "post": {
+        "tags": [
+          "PrivateStores"
+        ],
+        "operationId": "PrivateStore_GetOfferWithApprovalRequestsStatuses",
+        "description": "Get offer with approval request status foreach plan",
+        "parameters": [
+          {
+            "in": "body",
+            "name": "payload",
+            "schema": {
+              "$ref": "#/definitions/OfferDetailsProperties"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/PrivateStoreOfferWithRequestApproval"
+            }
+          },
+          "default": {
+            "description": "Microsoft.Marketplace error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "GetPrivateStoreOfferWithRequestApproval": {
+            "$ref": "./examples/GetPrivateStoreOfferWithRequestApproval.json"
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "PrivateStores"
+        ],
+        "operationId": "PrivateStore_CreateApprovalRequest",
+        "description": "Create approval request",
+        "parameters": [
+          {
+            "in": "body",
+            "name": "payload",
+            "schema": {
+              "$ref": "#/definitions/CreateApprovalRequestDetails"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/ApprovalRequestResponse"
+            }
+          },
+          "default": {
+            "description": "Microsoft.Marketplace error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "UpsertApprovalRequest": {
+            "$ref": "./examples/UpsertApprovalRequest.json"
+          }
+        }
+      }
+    },
+    "/providers/Microsoft.Marketplace/privateStores/{privateStoreId}/offers/{offerId}/adminRequestApprovals": {
```

Comment:
requestApprovals should not be modeled under multiple resource types with the same IDs and bodies.  If you want RBAC restrictions on who can approve a request make it an explicit action on the requestApprovals/{offerId} resource

Reply:
@pilor , please review the changes. thanks!